clippy_lints/src/format_args.rs

```diff
@@ -375,6 +381,21 @@ fn check_one_arg(
         fixes.push((pos_span, replacement));
         fixes.push((arg_span, String::new()));
         true  // successful inlining, continue checking
+    } else if !matches!(arg.kind, FormatArgumentKind::Captured(_))
+        && let Some(FormatPlaceholder{span: Some(placeholder_span), ..}) = placeholder
+        && let rustc_ast::ExprKind::Lit(lit) = &arg.expr.kind
+        && lit.kind == LitKind::Str // FIXME: lit.kind must match the format string kind
```

Comment:
This isn't clear to me. What other kinds would there be?

Reply:
there are also [raw strings](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals)

Reply:
And possibly byte strings too?  Not certain